### PR TITLE
Make August 2024 News Public

### PR DESCRIPTION
**Pull request by cticenhour**
@idaholab/moose-ccb As usual, please add your contributions from the month of August. 

Refs #11447

### CODE REVIEW

File: modules/doc/content/newsletter/2024/2024_08.md [hunk 1]
```diff
@@ -1,14 +1,16 @@
 # MOOSE Newsletter (August 2024)
 
-!alert! construction title=In Progress
-This MOOSE Newsletter edition is in progress. Please check back in September 2024
-for a complete description of all MOOSE changes.
-!alert-end!
-
 ## MOOSE Improvements
 
+The [SolutionInvalidInterface.md] now offers the option to create warnings that are tallied during a simulation,
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2024/2024_08.md:5]:
```suggestion
### Warnings and warning tallies now available for SolutionInvalidInterface

[SolutionInvalidInterface.md] now offers the option to create warnings that are tallied during a simulation,
```

File: modules/doc/content/newsletter/2024/2024_08.md [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+similar to the `solutionInvalid`-cases, but do not cause the solution to be rejected. Knowing the number of instances
+that a warning was triggered on is helpful for the modeler to decide manually whether the solution is acceptable.
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2024/2024_08.md:7]:
```suggestion
that a warning was triggered is helpful for the modeler to decide manually whether the solution is acceptable. 
```

File: modules/doc/content/newsletter/2024/2024_08.md [hunk 3]
```diff
@@ -27,4 +29,30 @@ for a complete description of all MOOSE changes.
...
+- The [Terminator.md] can now output messages without failing the time step or returning an error,
+  with its new 'NONE' option to the [!param](/UserObjects/Terminator/fail_mode) parameter.
+- `VectorNeumannBC` was renamed to [DirectionalNeumannBC.md] for clarity.
+- `MeshExtruderGenerator` was deprecated to prepare for its removal, as the [AdvancedExtruderGenerator.md]
+  offers all its functionalities and more.
+- `MaterialVectorPostprocessor` was renamed to [ElementMaterialSampler.md] to follow the naming convention
+  used for the other vectorpostprocessor sampler objects.
+- [TransientMultiApp.md] now always restore by default on the end of fixed point iterations. Not restoring
+  was an optimization that is not valid if there are stateful properties, or generally quantities computed on
+  `TIMESTEP_BEGIN`.
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2024/2024_08.md:58]:
```suggestion
- [TransientMultiApp.md] now always restores by default at the end of fixed point iterations. Not restoring
  was an optimization that is not valid if there are stateful properties or, generally, quantities computed on
  `TIMESTEP_BEGIN`.
```

File: modules/doc/content/newsletter/2024/2024_08.md [hunk 4]
```diff
@@ -1,14 +1,38 @@
...
-This MOOSE Newsletter edition is in progress. Please check back in September 2024
-for a complete description of all MOOSE changes.
-!alert-end!
+## OpenMPI added to Conda MOOSE Packages alongside MPICH
+
+You may now specify either OpenMPI or the MPICH MPI wrapper when installing MOOSE Conda packages by appending `openmpi` or `mpich` to the `conda create/install` line:
+```bash
+conda create -n moose moose-dev openmpi||mpich
+```
+*Note: OpenMPI being a new package addition, is a bit under-tested at the moment.*
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2024/2024_08.md:9]:
```suggestion

*Note: OpenMPI, being a new package addition, is a bit under-tested at the moment. It is recommended for use only by experienced users and developers!*
```

File: modules/doc/content/newsletter/2024/2024_08.md [hunk 5]
```diff
@@ -1,14 +1,38 @@
 # MOOSE Newsletter (August 2024)
 
-!alert! construction title=In Progress
-This MOOSE Newsletter edition is in progress. Please check back in September 2024
-for a complete description of all MOOSE changes.
-!alert-end!
+## OpenMPI added to Conda MOOSE Packages alongside MPICH
+
+You may now specify either OpenMPI or the MPICH MPI wrapper when installing MOOSE Conda packages by appending `openmpi` or `mpich` to the `conda create/install` line:
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2024/2024_08.md:5]:
```suggestion
You may now specify either OpenMPI or the MPICH MPI wrapper when installing MOOSE Conda packages by appending `openmpi` or `mpich` to the `conda create/install` line:

```

File: modules/doc/content/newsletter/2024/2024_08.md [hunk 6]
```diff
@@ -1,14 +1,38 @@
...
+[SolutionInvalidInterface.md] now offers the option to create warnings that are tallied during a simulation,
+similar to the `solutionInvalid`-cases, but do not cause the solution to be rejected. Knowing the number of instances
+that a warning was triggered is helpful for the modeler to decide manually whether the solution is acceptable.
+These warnings are created using the `flagSolutionWarning` routine.
+
+### Added fluid properties class for a two-phase fluid mixed with a non-condensable gas
+
+The [TwoPhaseNCGPartialPressureFluidProperties.md] class was added, which derives
+from `TwoPhaseNCGFluidProperties` and takes any `TwoPhaseFluidProperties` and
+`SinglePhaseFluidProperties` as input and creates an [IdealRealGasMixtureFluidProperties.md],
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2024/2024_08.md:24]:
```suggestion
from `TwoPhaseNCGFluidProperties`, takes any `TwoPhaseFluidProperties` and
`SinglePhaseFluidProperties` as input, and creates an [IdealRealGasMixtureFluidProperties.md],
```

**Comment by GiudGiud** [comment]
Squashed in the suggestion. I m not sure how disruptive that is to others editing. Please feel free to complain if that is actually messing with your rebases.

File: modules/doc/content/newsletter/2024/2024_06.md [hunk 7]
```diff
@@ -10,6 +10,12 @@ A new component, [FileMeshPhysicsComponent.md], was created in the thermal hydra
 
 The [WCNSFVTwoPhaseMixturePhysics.md] was added to the Navier Stokes module. This facilitates the setup of fully coupled finite volume simulations using a simple two phase mixture model. It can be used for both solidification and liquid mixture problems.
 
+### Mesh generator to perform mesh rotational extrusion
+
+The [RevolveMeshGenerator.md] was added, to create a revolution, or rotational extrusion, of a 2D mesh into 3D.
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2024/2024_06.md:15]:
```suggestion
The [RevolveMeshGenerator.md] was added to create a revolution, or rotational extrusion, of a 2D mesh into 3D.
```

File: modules/doc/content/newsletter/2024/2024_08.md [hunk 8]
```diff
@@ -1,14 +1,87 @@
...
+*Note: OpenMPI, being a new package addition, is a bit under-tested at the moment. It is recommended for use only by experienced users and developers!*
 
 ## MOOSE Improvements
 
+### New system: Correctors
+
+The [Corrector system](syntax/Correctors/index.md) was introduced to offer first-class support for modifying
+solution variables (also called nonlinear variables, except when using the linear finite volume discretization).
+This can be for re-normalization purposes, for shifting the entire variable, and so on. `Correctors` derive from
+`UserObjects`, they are simply specialized for this purpose and use the `[Corrector]` syntax.
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2024/2024_08.md:20]:
```suggestion
`UserObjects`; they are simply specialized for this purpose and use the `[Corrector]` syntax.
```

File: modules/doc/content/newsletter/2024/2024_08.md [hunk 9]
```diff
@@ -1,14 +1,87 @@
...
+We are curious to see how users seek to modify the output of the nonlinear solver, so please let us know how you use this new system!
+
+### New system: MeshModifiers
+
+The [MeshModifier system](syntax/MeshModifiers/index.md) was introduced to offer first-class support for modifying
+the mesh during a simulation. This can be to change the subdomain id of certain elements, to add or remove sides from a sideset, and so on.
+`MeshModifiers` derive from `UserObjects`, they are simply specialized for this purpose and use the `[MeshModifier]` syntax.
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2024/2024_08.md:27]:
```suggestion
`MeshModifiers` derive from `UserObjects`; they are simply specialized for this purpose and use the `[MeshModifier]` syntax.
```

File: modules/doc/content/newsletter/2024/2024_08.md [hunk 10]
```diff
@@ -1,14 +1,87 @@
...
+### Warnings and warning tallies now available for SolutionInvalidInterface
+
+[SolutionInvalidInterface.md] now offers the option to create warnings that are tallied during a simulation,
+similar to the `solutionInvalid`-cases, but do not cause the solution to be rejected. Knowing the number of instances
+that a warning was triggered is helpful for the modeler to decide manually whether the solution is acceptable.
+These warnings are created using the `flagSolutionWarning` routine.
+
+### Block restriction debug output
+
+Block restriction of objects can now be output using an option in the [Debug block](SetupDebugAction.md).
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2024/2024_08.md:44]:
```suggestion
Block restriction of objects can now be output using the [!param](/Debug/SetupDebugAction/show_block_restriction) parameter in the [Debug block](SetupDebugAction.md).
```

File: modules/doc/content/newsletter/2024/2024_08.md [hunk 11]
```diff
@@ -1,14 +1,87 @@
...
+
+### MOOSE server improvements
+
+These improvements will be available on the next update of the VSCode MOOSE plugin.
+
+- When writing a new object in an input, the required parameters are automatically added to the input file. The user still has to fill in their value.
+- `!include` directives are no longer removed by automatic formatting
+- Inputs with a broken format will no longer be subject to automatic formatting, which would not improve the broken format
+- Parameters with a specific string type, for example `FunctionName`, will now automatically suggest the valid inputs from the rest of the input file, for example the list of `Functions` already specified in the input
```
Comment (cticenhour) [inline modules/doc/content/newsletter/2024/2024_08.md:53]:
```suggestion
- `!include` directives are no longer removed by automatic formatting.
- Inputs with a broken format will no longer be subject to automatic formatting, which would not improve the broken format.
- Parameters with a specific string type, for example `FunctionName`, will now automatically suggest the valid inputs from the rest of the input file. In this example, the plugin might suggest the list of `Functions` already specified in the input.
```

**Comment by roystgnr** [comment]
> Please feel free to complain if that is actually messing with your rebases.

And if so, you'll un-force-push?  :confused:

#28508 is getting hit by those schemadiff failures anyway, so whenever that can be expected to pass I'll make a serious try to rebase afterward and see how it goes.  I'm already getting merge conflicts when I try, though.

**Comment by GiudGiud** [comment]
conflicts are expected given that you are adding to the same file. 
but is the commit history actually getting wrecked to the point you have to interactively rebase and pick the right commits?

**Comment by roystgnr** [comment]
It's trying to reapply two of the commits that I rebased off of, both of which conflict heavily, but if I just remove those interactively then it's a clean rebase.

**Review by GiudGiud — APPROVED** [review]

**Pull request merged**